### ISSUE TITLE
PropTypes are not removed from inherited classes

### ISSUE THREAD

**Issue by azasypkin** [oliviertassinari/babel-plugin-transform-react-remove-prop-types#29]
Take a look at `Parent` class [1] and `Child` class [2] that extends `Parent`.

Once it's compiled with babel with `babel-plugin-transform-react-remove-prop-types` plugin, PropTypes are removed from `Parent` [3] class, but not from `Child` [4].

The test case repo contains gulpfile and .babelrc as well, @oliviertassinari.

Thanks!

[1] https://github.com/azasypkin/issues/blob/master/babel-plugin-transform-react-remove-prop-types/parent.jsx#L9-L11
[2] https://github.com/azasypkin/issues/blob/master/babel-plugin-transform-react-remove-prop-types/child.jsx#L10-L13
[3] https://github.com/azasypkin/issues/blob/master/babel-plugin-transform-react-remove-prop-types/dist/parent.js
[4] https://github.com/azasypkin/issues/blob/master/babel-plugin-transform-react-remove-prop-types/dist/child.js

**Comment by oliviertassinari** [comment]
Yeah, that's actually a design decision, have a look at https://github.com/oliviertassinari/babel-plugin-transform-react-remove-prop-types/pull/8
> It's also taking a side by only accepting `React.Component` and `Component` as class names.
Anything else, like `BaseComponent` won't be optimised.

**Comment by azasypkin** [comment]
Ah, I see, thanks for the pointer!

So does it mean that Babel doesn't allow to traverse up the inheritance chain at all or it's just current limitation of the plugin?

**Comment by oliviertassinari** [comment]
The thing is, I can't remove the `propTypes` from any class. I should only do it on a React class component.
I need a heuristic to do so. I relying on the `React.Component` and `Component` class names.
That works great in +90% of the cases.
For your specific cases, we could also be looking for a render method and assert that this method is returning a react element.

**TL;DR** The quick win is to rename your `Parent` class. The right way to fix this issue would be to improve the heuristic.
Notice that using inheritance is considered a bad pattern in React. I would rather use composition.

**Comment by oliviertassinari** [comment]
This issue is quite old. I haven't seen more need for it. I'm closing it. Thanks for raising your voice.
